src/core/server/saved_objects/permission_control/routes/status.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AuthStatus, InternalHttpServiceSetup, IRouter } from '../../../http';
+
+/**
+ * router to handle permission control settings
+ */
+export const registerStatusRoute = (router: IRouter, http: InternalHttpServiceSetup) => {
+  router.get(
+    {
+      path: '/status',
+      validate: {},
+    },
+    router.handleLegacyErrors(async (context, req, res) => {
+      // For now, we consider permission control is enabled when `auth` interceptor is registered
+      const enabled = http.auth.get(req).status !== AuthStatus.unknown;
```

Comment:
Could we create a separate util function to share the request judgement logic?

Reply:
I think that makes sense, but since it's not used in too many places and also the one line logic is pretty straight-forward, I will leave it for now.